### PR TITLE
Raise an error for ifs in comprehension statements

### PR DESCRIPTION
**Pull request by EmilyBourne**
Raise an error when if conditions are used in comprehension statements.
E.g:
```python
a = [i for i in range(10) if i%2==0]
```
See #1410

### CODE REVIEW

File: pyccel/parser/syntactic.py [hunk 1]
```diff
@@ -1119,6 +1119,10 @@ def _visit_comprehension(self, stmt):
 
         self.exit_loop_scope()
 
+        if stmt.ifs:
+            errors.report("Can't calculate list length when comprehension contains if statements.\n" + PYCCEL_RESTRICTION_TODO,
```
Comment (bauom) [inline pyccel/parser/syntactic.py:1123]: I found this error a bit confusing, the error assumes that i know the issue.
(i did understand it after I read the issue)

Comment (EmilyBourne) [reply pyccel/parser/syntactic.py:1123]: Hmm, I was hoping that the line number would help make it obvious. Do you have another suggestion?

Comment (EmilyBourne) [reply pyccel/parser/syntactic.py:1123]:
```suggestion
            errors.report("Cannot handle if statements in list comprehensions. List length cannot be calculated.\n" + PYCCEL_RESTRICTION_TODO,
```
?

Comment (bauom) [reply pyccel/parser/syntactic.py:1124]: I think This describe the Problem clearly. `Cannot handle if statements in list comprehensions.`
which for me was hard to know from the start with the previous message.

Comment (EmilyBourne) [reply pyccel/parser/syntactic.py]: Fixed

**Review by bauom — APPROVED** [review]

**Pull request merged**